Content.Server/Clothing/EntitySystems/ToggleableClothingSystem.cs

```diff
@@ -0,0 +1,174 @@
+        {
+            // self toggled clothing, like masks
+            if (component.SelfToggle)
+            {
+                OnSelfToggleClothing(uid, component, args);
+                args.Handled = true;
+                return;
+            }
+
+            // continue on for standard toggleable clothing, like hardsuits
+            if (args.Handled || component.Container == null || component.ClothingUid == null)
+                return;
+
+            var parent = Transform(uid).ParentUid;
+            if (component.Container.ContainedEntity == null)
+                _inventorySystem.TryUnequip(parent, component.Slot);
+            else if (_inventorySystem.TryGetSlotEntity(parent, component.Slot, out var existing))
+            {
+                _popupSystem.PopupEntity(Loc.GetString("toggleable-clothing-remove-first", ("entity", existing)),
+                    args.Performer, Filter.Entities(args.Performer));
+            }
+            else
+                _inventorySystem.TryEquip(parent, component.ClothingUid.Value, component.Slot);
+
+            args.Handled = true;
+        }
+
+        private void OnSelfToggleClothing(EntityUid uid, ToggleableClothingComponent component, ToggleClothingEvent args)
+        {
+            if (args.Handled)
+                return;
+
+            switch (component.Slot)
+            {
+                case "mask":
+                    if (TryEquip(uid, component, args, component.Slot))
+                    {
+                        component.IsToggled ^= true;
+                        ToggleMask(uid, component, args.Performer);
+                        //TODO: update visuals?
+                    }
+                    break;
+            }
+        }
+
+        // reset on pickup to ensure not initially equipped in toggled state
+        private void ResetSelfToggleClothing(EntityUid uid, ToggleableClothingComponent component)
+        {
+            switch (component.Slot)
+            {
+                case "mask":
+                    component.IsToggled = false;
+                    ToggleMask(uid, component, default, false);
+                    break;
+            }
+        }
+
+        // On toggle, try to equip if not already equipped
+        private bool TryEquip(EntityUid uid, ToggleableClothingComponent component, ToggleClothingEvent args, string slot)
+        {
+            if (_inventorySystem.TryGetSlotEntity(args.Performer, slot, out var existing) && !component.Owner.Equals(existing))
+            {
+                _popupSystem.PopupEntity(Loc.GetString("toggleable-clothing-remove-first", ("entity", existing)),
+                                    args.Performer, Filter.Entities(args.Performer));
+                return false;
+            }
+            else if (existing == null)
+            {
+                _inventorySystem.TryEquip(args.Performer, component.Owner, slot);
+                return false;
+            }
+
+            return true;
+        }
```

Comment:
just call `_inventorySystem.TryEquip(args.Performer, component.Owner, slot);`, it already does all this logic